### DISCUSSION TITLE
Workflow Designer: Elsa Studio

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [Blazor-Diagrams/Blazor.Diagrams#334] [Show and tell]
https://github.com/elsa-workflows/elsa-core/issues/3810#issuecomment-1650643045

Implemented using Blazor Diagram!

![msedge_YEZIkQaFYo](https://github.com/Blazor-Diagrams/Blazor.Diagrams/assets/49812372/f64fc8d8-e2cb-46ad-874e-ff7d584a38c2)

**Comment by zHaytam** [comment]
Oh my god, that's great!
Do you have any issues or things you wished were in the library during your implementation?

**Reply by GeorgeS2019** [reply]
@zHaytam 
It seems much of the UI framework is blazor, [the diagram part remains typescript](https://github.com/elsa-workflows/elsa-studio/issues/1#issuecomment-1656737823).

Ideally, we need to figure out a reusable module to replace the typescript diagram with blazor, making the Elsa Studio fully 100% c#

**Reply by GeorgeS2019** [reply]
@zHaytam 

Here is the [discussion](https://github.com/elsa-workflows/elsa-studio/discussions/10#discussion-5466426)

**Comment by GeorgeS2019** [comment]
@zHaytam 
All these Javascript interopt is very slow. 
It would be great if there is a Blazor Diagram based workflow designer, a separate project to what you have done with database designer.

https://workflowengine.io/documentation/workflow-designer-in-blazor-application
![image](https://github.com/Blazor-Diagrams/Blazor.Diagrams/assets/49812372/bb2da7b7-0d67-40e9-8518-ac6a586efb88)
